projects/ngneat/loadoff/src/lib/retainResult.ts

```diff
@@ -0,0 +1,20 @@
+import { OperatorFunction, pipe } from 'rxjs';
+import { scan, startWith } from 'rxjs/operators';
+import { AsyncState, createAsyncState } from './toAsyncState';
+
+/**
+ * Operator that retains the previous result on a new emission of an AsyncState item.
+ * This can be useful for implementing a refresh mechanism, where you want to keep displaying the old value while fetching a new value
+ */
+export function retainResult<T, E = any>(): OperatorFunction<AsyncState<T, E>, AsyncState<T, E>> {
+  return pipe(
+    startWith(createAsyncState<T, E>()),
```

Comment:
Do you think the startWith should be here or let the consumer define it?

Reply:
We could add an optional startWithValue param. Something like the code below, what do you think?

```
export function retainResult<T, E = any>(startWithValue: AsyncState<T, E> = createAsyncState<T, E>()): OperatorFunction<AsyncState<T, E>, AsyncState<T, E>> {
  return pipe(
    startWith(startWithValue),
```